src/theme/components/menu.ts

```diff
@@ -0,0 +1,146 @@
+import { SectionTypes } from "../../helpers/types";
+import { getSectionColors } from "../../utils/getSectionColors";
+interface MenuStyleProps {
+  highlightColor: SectionTypes;
+  showBorder: boolean;
+  showLabel: boolean;
+}
+
+const Menu = {
+  parts: [
+    "groupItem",
+    "actionItem",
+    "dividerItem",
+    "selected",
+    "menuButton",
+    "menuGroup",
+    "menuList",
+  ],
+  baseStyle: ({
+    highlightColor = "blogs",
+    showBorder,
+    showLabel,
+  }: MenuStyleProps) => {
+    const sectionColor = getSectionColors({
+      type: highlightColor,
+      colorValue: highlightColor === "blogs" ? "secondary" : "primary",
+    });
+    const lightSectionColor = sectionColor + "-05";
+    const darkSectionColor = "dark." + sectionColor + "-10";
+    return {
+      groupItem: {
+        padding: "8px 12px",
+        lineHeight: "150%",
+        fontWeight: "510",
+        fontSize: "12px",
+        margin: "0px",
+        _dark: {
+          textColor: "ui.white",
+          bg: "dark.ui.bg.default",
+        },
+      },
+      dividerItem: {
+        margin: "0px",
+        _dark: {
+          textColor: "ui.white",
+          bg: "dark.ui.bg.default",
+        },
+      },
+      actionItem: {
+        outline: "none !important",
+        padding: "8px 12px",
+        fontSize: "14px",
+        lineHeight: "150%",
+        _hover: {
+          fontWeight: 510,
+          bg: "ui.bg.hover",
+        },
+        _focus: {
+          fontWeight: 510,
+          bg: "ui.bg.hover",
+        },
+        _dark: {
+          textColor: "ui.white",
+          bg: "dark.ui.bg.default",
+          "> div > svg": {
+            fill: "ui.white",
+          },
+          _hover: {
+            bg: "dark.ui.bg.active",
+          },
+        },
+      },
+      selected: {
+        _hover: {
+          background:
+            highlightColor === "blogs" ? "ui.bg.default" : lightSectionColor,
+        },
+        _active: {
+          background:
+            highlightColor === "blogs" ? "ui.bg.default" : lightSectionColor,
+        },
+        fontWeight: 500,
+        borderLeftColor:
+          highlightColor === "whatsOn" ? "#E9E9E933" : sectionColor,
+        textColor: highlightColor === "blogs" ? "ui.black" : sectionColor,
+        borderWidth: "0px 0px 0px 2px",
+        background:
+          highlightColor === "blogs" ? "ui.bg.default" : lightSectionColor,
+        _dark: {
+          textColor: "ui.white",
+          background:
+            highlightColor === "whatsOn"
+              ? "dark.ui.bg.hover"
+              : darkSectionColor,
+        },
+      },
+      menuButton: {
```

Comment:
For general formatting, add the following to `menuButton`:
```
minHeight: { base: defaultElementSizes.mobileFieldHeight, md: "40px" },
minWidth: { base: defaultElementSizes.mobileFieldHeight, md: "40px" },
whiteSpace: "nowrap",
```

And for the icon fix, add this:
```
"> span ": {
  alignItems: showLabel ? undefined : "center",
  display: showLabel ? undefined : "flex",
  justifyContent: showLabel ? undefined : "center",
},
```